core/src/test/scala/org/apache/spark/scheduler/SchedulerIntegrationSuite.scala

```diff
@@ -133,6 +133,11 @@ abstract class SchedulerIntegrationSuite[T <: MockBackend: ClassTag] extends Spa
       // when the job succeeds
       assert(taskScheduler.runningTaskSets.isEmpty)
       assert(!backend.hasTasks)
+    } else {
+      // Note that we CANNOT check for empty results on a failure -- the resultHandler will
```

Comment:
It's hard to correlate this comment with the code around it. It seems like it belongs in the call site, where the `results` list was actually being checked (and you removed the assert). But without the assert, the comment would be out of place there too.

I just find it very improbable that someone will read this comment when writing a test. It would help if, instead, the expected results were passed to this method, so that the asserts happen here instead of in the body of every test. Then the comment is fine where it is, because this method would handle the asserts (or lack thereof).

Also, the same assert exists in the "job failure after 4 attempts" test in `SchedulerIntegrationSuite`. Doesn't that need to change too?